Examples/Algorithms/Geant4/include/ActsExamples/DDG4/DDG4DetectorConstruction.hpp

```diff
@@ -28,7 +29,9 @@ struct DD4hepDetector;
 /// Construct the Geant4 detector from a DD4hep description.
 class DDG4DetectorConstruction final : public G4VUserDetectorConstruction {
  public:
-  DDG4DetectorConstruction(std::shared_ptr<DD4hep::DD4hepDetector> detector);
+  DDG4DetectorConstruction(
+      std::shared_ptr<DD4hep::DD4hepDetector> detector,
+      std::vector<std::shared_ptr<RegionCreator>> regionCreators);
```

Comment:
I wonder if sharing these has a benefit or if we can just do `std::vector<RegionCreator>`

Reply:
It has `std::unique_ptr` members so they are not trivially copiable. Other complex structures that come from python are also stored in shared pointers and I just followed the same conventions.

I'm also not sure what the lifetime relation between the factory and the wrapper class is to pass references/pointers around.